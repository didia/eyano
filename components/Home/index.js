import Layout from '../Layout';

const Home = () => <Layout>Welcome to the home page</Layout>;

export default Home;
