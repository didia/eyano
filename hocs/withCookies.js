// Vendor
import { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { CookiesProvider } from 'react-cookie';
import { loadGetInitialProps } from 'next/dist/lib/utils';

export default ComposedComponent =>
  class WithCookies extends PureComponent {
    static propTypes = {
      cookies: PropTypes.object
    };

    static async getInitialProps({ req }) {
      return {
        cookies: req ? req.universalCookies : undefined,
        ...loadGetInitialProps(ComposedComponent)
      };
    }

    render() {
      const componentProps = { ...this.props };
      delete componentProps.cookies;

      return (
        <CookiesProvider cookies={this.props.cookies}>
          <ComposedComponent {...componentProps} />
        </CookiesProvider>
      );
    }
  };
