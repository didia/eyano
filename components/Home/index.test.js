import React from 'react';
import renderer from 'react-test-renderer';
import Home from './index';
import withCookies from '../../hocs/withCookies';

describe('Home Component', () => {
  it('renders as expected', () => {
    const component = renderer.create(withCookies(<Home />));

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
