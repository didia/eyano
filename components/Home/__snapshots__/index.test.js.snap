// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders as expected 1`] = `null`;
